R.swift/Generators/ResourceGenerator.swift

```diff
@@ -31,19 +31,33 @@ private struct GeneratorResults {
   }
 }
 
-func generateResourceStructsWithResources(resources: Resources, bundleIdentifier: String) -> (Struct, Struct) {
+func generateResourceStructsWithResources(resources: Resources, bundleIdentifier: String, targetType: TargetType) -> (Struct, Struct) {
 
-  let generators: [Generator] = [
-      ImageGenerator(assetFolders: resources.assetFolders, images: resources.images),
+  let generators: [Generator]
+  switch targetType {
+  case .iOS, .tvOS:
+    generators = [
+      ImageGenerator(assetFolders: resources.assetFolders, images: resources.images, withTrait: true),
       ColorGenerator(colorPalettes: resources.colors),
       FontGenerator(fonts: resources.fonts),
       SegueGenerator(storyboards: resources.storyboards),
       StoryboardGenerator(storyboards: resources.storyboards),
       NibGenerator(nibs: resources.nibs),
       ReuseIdentifierGenerator(reusables: resources.reusables),
       ResourceFileGenerator(resourceFiles: resources.resourceFiles),
-      StringsGenerator(localizableStrings: resources.localizableStrings),
+      StringsGenerator(localizableStrings: resources.localizableStrings)]
+  case .watchOS:
```

Comment:
I would reserve this switch only for thing where it's absolutely necessary to differentiate between watchOS and iOS. For example; Images look exactly the same, so the generated doesn't know whether it must generate an image for watchOS or iOS, but the code must be different.

For a storyboard for example it doesn't matter, the generator simply won't encounter iOS classes in a watchOS storyboard, so it will automatically generate the correct imports and use te correct classes. Should never be a problem, so we don't have to use seperate code paths for them.